src/riot-rs-threads/src/lib.rs

```diff
@@ -322,6 +360,24 @@ pub fn wakeup(thread_id: ThreadId) -> bool {
     })
 }
 
+/// Returns the priority of a thread.
+///
+/// Returns `None` if this is not a valid thread.
+pub fn get_priority(thread_id: ThreadId) -> Option<u8> {
+    THREADS.with_mut(|threads| threads.get_priority(thread_id).map(|rq| *rq))
+}
+
+/// Changes the priority of a thread.
+///
+/// This might trigger a context switch.
+pub fn set_priority(thread_id: ThreadId, prio: u8) {
+    THREADS.with_mut(|mut threads| {
+        if threads.set_priority(thread_id, RunqueueId::new(prio)) {
+            schedule();
+        }
+    })
+}
+
```

Comment:
I used `u8` instead of `RunqueueId` in the API here to match the `u8` used in the `thread_create*` functions, but maybe we should use `RunqueueId` in our API instead>

Reply:
> Are all possible values of `RunqueueId` valid priorities?

Unfortunately not. `RunqueueId` is an u8 and thus can be up to 255, but:
1. `riot_rs_threads::SCHED_PRIO_LEVELS` can be lower than that, which is the max # of prio levels we support
2. "runqueue numbers fit in usize bits (supporting max 32 priority levels)": see https://github.com/future-proof-iot/RIOT-rs/pull/414#discussion_r1758503064